File: exercism/python/space-age/space_age.py
plants_years = {
    'mercury':  0.2408467,
    'venus':  0.61519726,
    'mars': 1.8808158,
    'jupiter': 11.862615,
    'saturn': 29.447498,
    'uranus': 84.016846,
    'neptune': 164.79132,
    'earth': 1
}


class SpaceAge(object):
    def __init__(self, seconds):
        self.seconds = seconds
        for planet in plants_years:
            self.add_function(planet)

    def add_function(self, planet):
        setattr(self, 'on_' + planet,
                lambda: self.calculate_years(planet))

    def calculate_years(self, planet):
        return round(self.seconds / 31557600 / plants_years[planet], 2)

File: exercism/scala/space-age/src/main/scala/SpaceAge.scala
object SpaceAge {
  val earthOrbitalPeriod: Double = 31557600
  val plantsYears = Map[String, Double](
    "mercury" ->  0.2408467,
    "venus" ->  0.61519726,
    "mars" -> 1.8808158,
    "jupiter" -> 11.862615,
    "saturn" -> 29.447498,
    "uranus" -> 84.016846,
    "neptune" -> 164.79132,
    "earth" -> 1
  ) 

  def calcYearsFor(seconds: Double, plant: String): Double = seconds / earthOrbitalPeriod / plantsYears(plant)

  def onEarth(seconds: Double): Double = calcYearsFor(seconds, "earth")
  def onMercury(seconds: Double): Double = calcYearsFor(seconds, "mercury")
  def onVenus(seconds: Double): Double =  calcYearsFor(seconds, "venus")
  def onMars(seconds: Double): Double =   calcYearsFor(seconds, "mars")
  def onJupiter(seconds: Double): Double = calcYearsFor(seconds, "jupiter")
  def onSaturn(seconds: Double): Double =  calcYearsFor(seconds, "saturn")
  def onUranus(seconds: Double): Double =  calcYearsFor(seconds, "uranus")
  def onNeptune(seconds: Double): Double = calcYearsFor(seconds, "neptune") 

}
